app/rtc_compare.py

```diff
@@ -0,0 +1,329 @@
+            return np.array_equal(val_1, val_2, equal_nan=True)
+        else:
+            return np.array_equal(val_1, val_2)
+
+    elif len(shape_val_1)==1 and len(shape_val_2)==1:
+        # 1d vector
+        if 'shape' in dir(val_1[0]):
+            if isinstance(val_1[0], np.void):
+                list_val_1 = list(itertools.chain.from_iterable(val_1))
+                val_1_new = [None] * len(list_val_1)
+                for i_val, element_1 in enumerate(list_val_1):
+                    if isinstance(element_1, h5py.h5r.Reference):
+                        val_1_new[i_val] = hdf5_obj_1[element_1]
+                    else:
+                        val_1_new[i_val] = element_1
+
+                val_1 = val_1_new
+
+            elif (len(val_1[0].shape) == 1) and (isinstance(val_1[0][0], h5py.h5r.Reference)):
+                list_val_1 = list(itertools.chain.from_iterable(val_1))
+                val_1_new=[None] * len(list_val_1)
+                for i_val, element_1 in enumerate(list_val_1):
+                    if isinstance(element_1, h5py.h5r.Reference):
+                        val_1_new[i_val] = hdf5_obj_1[element_1]
+                    else:
+                        val_1_new[i_val] = element_1
+                val_1 = val_1_new
+
+        if 'shape' in dir(val_2[0]):
+            if isinstance(val_2[0], np.void):
+                list_val_2 = list(itertools.chain.from_iterable(val_2))
+                val_2_new = [None] * len(list_val_2)
+                for i_val, element_2 in enumerate(list_val_2):
+                    if isinstance(element_2, h5py.h5r.Reference):
+                        val_2_new[i_val] = hdf5_obj_2[element_2]
+                    else:
+                        val_2_new[i_val] = element_2
+
+                val_2 = val_2_new
+
+            elif (len(val_2[0].shape) == 1) and (isinstance(val_2[0][0], h5py.h5r.Reference)):
+                list_val_2 = list(itertools.chain.from_iterable(val_2))
+                val_2_new = [None]*len(list_val_2)
+                for i_val, element_2 in enumerate(list_val_2):
+                    if isinstance(element_2, h5py.h5r.Reference):
+                        val_2_new[i_val] = hdf5_obj_2[element_2]
+                    else:
+                        val_2_new[i_val] = element_2
+
+                val_2 = val_2_new
+
+        if isinstance(val_1, list) and isinstance(val_2, list):
+            if len(val_1) == len(val_2):
+                for id_element, element_1 in enumerate(val_1):
+                    element_2 = val_2[id_element]
+                    if element_1.shape == element_2.shape:
+                        if not np.allclose(element_1,
+                                           element_2,
+                                           RTC_S1_PRODUCTS_ERROR_TOLERANCE,
+                                           equal_nan=True):
+                            return False
+                    else:
+                        return False
```

Comment:
```suggestion
                    element_2 = val_2[id_element]
                    if element_1.shape != element_2.shape:
                        return False
                    if not np.allclose(element_1,
                                       element_2,
                                       RTC_S1_PRODUCTS_ERROR_TOLERANCE,
                                       equal_nan=True):
                        return False
```
Early return, whenever possible